src/main/java/com/mitchej123/hodgepodge/mixins/TargetedMod.java

```diff
@@ -7,7 +7,8 @@ public enum TargetedMod {
     FASTCRAFT("FastCraft", "fastcraft.Tweaker"),
     COFH_CORE("CoFHCore", "cofh.asm.LoadingPlugin", "CoFHCore"),
     THAUMCRAFT("Thaumcraft", null, "Thaumcraft"), // "thaumcraft.codechicken.core.launch.DepLoader"
-    GT5U("GregTech5u", null, "gregtech"),
+    GT5U("GregTech5u", null, "gregtech"), // Also matches GT6.
+    GT6API("GregTech6-API", null, "gregapi"), // Can be used to exclude GT6 from the GT5U target.
```

Comment:
I'd say also put in the coremod ID, so that it can be used for early mod checks.  

If this still runs in the full pack with the mixin set to late then great; otherwise I'd suggest moving it back to early.  

Note to self: Should look into parsing out the mods, without loading them, to get mod ids that will be loaded later... and possibly the ability to load them early if needed for other early mixins.

Reply:
I'm not able to test in the full pack because there is no release available that uses GTNHMixins, and adding it to the latest client pack (2.2.3) causes a crash. I only tested with GT5u/GT6 and IC2 on their own (plus the deps and Hodgepodge).

Moving it back to the early phase would essentially put us back at square one (crash if there's no GT). I could however move the GT5 check inside the mixin code (and turn it into an `@Inject`).

I will add the coremod ID, good idea.

Reply:
Could it be a similar approach to isGt5uLoaded() in the mixin code itself (similar to pollution)?

I can try this as is later this afternoon; my concern is doing ic2 asm/mixins in the late stage might be too late... but that'll depend on when those classes get loaded.

Reply:
longer term identifying mods before loading them is the approach I'll work on, but there's a few shortterm ideas here

Reply:
As is, if we moved it back to early, it'd not crash with GT6, but would otherwise crash without gt5u.

Reply:
> Could it be a similar approach to isGt5uLoaded() in the mixin code itself (similar to pollution)?

Yeah, that's what I meant, using a shim and wrapping it in a GT5 check. I already have stashed code that does that from when I was trying stuff, I can push it if it sounds good.

```java
    @Inject(method = "hasCompleteHazmat", at = @At("HEAD"), cancellable = true)
    private static void hasCompleteHazmat(EntityLivingBase entity, CallbackInfoReturnable<Boolean> cir) {
        if (Compat.isGT5Present()) {
            cir.setReturnValue(Compat.isWearingFullRadioHazmatGT5(entity));
        }
    }
```

I went with the late mixin approach for the PR to avoid the overhead this might cause, but you're right that it would be less likely to break things.

Reply:
Imo you should keep the current method *if* it works. Testing only with IC2 + GT5U (and with GT6) should be fine, if you still want to test with the full pack, rot13 explained what to replace [here](https://discord.com/channels/181078474394566657/603348502637969419/1040768009205198868).

Reply:
Nice, got it running. And the (late) mixin works.

<details>
<summary>Exact changes I made</summary>

* Switched from SpongeMixins to GTNHMixins 2.1.1
* Updated GalactiCraft to 3.0.47-GTNH (3.0.52-GTNH doesn't work due to an unrelated issue)
* Updated IFU to 1.9.5
* Updated Visual Prospecting to 1.0.36
* Updated AppleCore to 3.2.3
* Updated Hodgepodge to 2.0.8-3-gd435362.dirty (only change from d435362 is I added a debug print to see if the mixin runs)
</details>